extensions/khr/GLSL_KHR_cooperative_matrix.txt

```diff
@@ -0,0 +1,396 @@
+Name
+GL_KHR_cooperative_matrix
+
+Contact
+Neil Hickey, Arm (neil.hickey 'at' arm.com)
+
+Contributors
+Jeff Bolz, NVIDIA (jbolz 'at' nvidia.com)
+Kevin Petit, Arm (kevin.petit 'at' arm.com)
+
+Notice
+Copyright (c) 2022 The Khronos Group Inc. Copyright terms at http://www.khronos.org/registry/speccopyright.html
+
+Status
+Not yet released
+
+Version
+Last Modified: Feb 10, 2023
+Revision: 3
+
+
+
+Dependencies
+This extension can be applied to OpenGL GLSL versions 4.50
+(#version 450) and higher.
+This extension can be applied to OpenGL ES ESSL versions 3.20
+(#version 320) and higher.
+All these versions map GLSL/ESSL semantics to the same SPIR-V 1.5 semantics (approximating the most recent versions of GLSL/ESSL).
+
+This extension interacts with physical_storage_buffer,EXT_shader_explicit_arithmetic_types
+
+Overview
+
+This extension adds a new set of types known as "cooperative matrix" types,
+where the storage for and computations performed on the matrix are spread
+across a set of invocations such as a subgroup. These types give the
+implementation freedom in how to optimize matrix multiply operations.
+
+This extension introduces the types and built-in functions, but does not
+specify rules about what sizes/combinations are valid. This is left to
+the Vulkan extension specifications, and it is expected that different
+implementations may support different sizes. To help accommodate this,
+the dimensions of the cooperative types are parameterized and can be
+specialized via specialization constants.
+
+This extension introduces limited support for parameterized types, with
+the parameters specified as in C++ template syntax. The new built-in type
+"coopmat" is the only type that can be parameterized, and its parameters
+are a scalar type for the component type, an integer value that controls
+the scope of the type, the number of rows and columns in the matrix, and
+which matrix accumulation argument the matrix can be.
+
+Cooperative matrix types are only supported in certain shader stages, and
+the supported stages can be queried from the API. There are no compile-time
+checks to disallow cooperative matrix types in any shader stage.
+
+Mapping to SPIR-V
+-----------------
+
+For informational purposes (non-normative), the following is an
+expected way for an implementation to map GLSL constructs to SPIR-V
+constructs:
+
+coopmat -> OpTypeCooperativeMatrixKHR
+coopmat constructor from scalar value -> OpConstantComposite
+coopmat constructor from coopmat -> Op*Convert, OpConvert*To*
+coopmat.length() -> OpCooperativeMatrixLength
+coopmat[i] -> OpCompositeExtract/OpCompositeInsert/OpAccessChain
++, -, *, / -> OpFAdd, OpFNegate/OpFSub, OpMatrixTimesScalar, OpFDiv
```

Comment:
We might want to list **OpFMul** as well. Also, is it worth adding the mapping for integer operations?